private/pkg/oauth2/client_test.go

```diff
@@ -106,7 +106,9 @@ func TestRegisterDevice(t *testing.T) {
 			})
 			output, err := c.RegisterDevice(ctx, test.input)
 			assert.Equal(t, test.output, output)
-			assert.Equal(t, err, test.err)
+			if test.err != nil {
+				assert.EqualError(t, err, test.err.Error())
+			}
```

Comment:
adjusted these tests slightly; using `Equal` on a joined error doesn't work in the same way. It seemed more important for these tests to assert that the output matches.